src/libs/DebugUtils.ts

```diff
@@ -355,8 +378,8 @@ function validateArray(
 /**
  * Validates if a string is a valid representation of an object.
  */
-function validateObject(value: string, type: ObjectType, collectionIndexType?: 'string' | 'number') {
-    if (value === 'undefined') {
+function validateObject<T extends Record<string, unknown>>(value: string, type: ObjectType<T>, collectionIndexType?: 'string' | 'number') {
```

Comment:
Did you update the unit tests to cover the changes?

Reply:
I did not but I'll confirm if it's needed, thanks! 😄